typeof keyword in struct filed produces c error

### Describe the bug

While implementing keywords support in intellij-v I noticed using `typeof` as identifier in structs produces C error. I'm not sure if it should be supported or should be treated as error but definetly it should not produce c error.

### Reproduction Steps

```v
struct Keywords {
    typeof     u8 // c error
}
```

### Expected Behavior

Either be allowed to use the keyword or produce an error during code check.

### Current Behavior

C error:

```
======== Output of the C Compiler (/home/jakub/programy/v/thirdparty/tcc/tcc.exe) ========
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c:1164: error: '(' expected (got ";")
==========================================================================================
======== Output of the C Compiler (cc) ========
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c:1164:12: error: expected identifier or ‘(’ before ‘typeof’
 1164 |         u8 typeof;
      |            ^~~~~~
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c:1165:1: warning: no semicolon at end of struct or union
 1165 | };
      | ^
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c: In function ‘main__main’:
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c:13606:50: error: expected identifier before ‘typeof’
13606 |         main__Keywords test = ((main__Keywords){.typeof = 0,});
      |                                                  ^~~~~~
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c:13606:50: warning: excess elements in struct initializer
/home/jakub/repos/intellij-v/tmp/struct keyword fields.tmp.c:13606:50: note: (near initialization for ‘(anonymous)’)
===============================================
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V full version: V 0.4.8 5c65e58.399159f

### Environment details (OS name and version, etc.)

```V full version: V 0.4.8 5c65e58.399159f
OS: linux, Linux Mint 21.3
Processor: 20 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i9-12900H

getwd: /home/jakub/repos/intellij-v
vexe: /home/jakub/programy/v/v
vexe mtime: 2024-11-06 11:13:52

vroot: OK, value: /home/jakub/programy/v
VMODULES: OK, value: /home/jakub/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.34.1
Git vroot status: 0.4.8-287-g399159ff (4 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
emcc version: N/A
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzJiOWYxM2U3Y2M1YTc4NTQyOGYwZTkiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.baeTlhGN1geiYE9FWJDkZN0YbNUkUv535Szs7IlK4GM">Huly&reg;: <b>V_0.6-21225</b></a></sub>

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -25,8 +25,8 @@ const c_reserved = ['asm', 'array', 'auto', 'bool', 'break', 'calloc', 'case', '
 	'exit', 'export', 'extern', 'false', 'float', 'for', 'free', 'goto', 'if', 'inline', 'int',
 	'link', 'long', 'malloc', 'namespace', 'new', 'nil', 'panic', 'register', 'restrict', 'return',
 	'short', 'signed', 'sizeof', 'static', 'string', 'struct', 'switch', 'typedef', 'typename',
-	'union', 'unix', 'unsigned', 'void', 'volatile', 'while', 'template', 'true', 'small', 'stdout',
-	'stdin', 'stderr', 'far', 'near', 'huge', 'requires']
+	'typeof', 'union', 'unix', 'unsigned', 'void', 'volatile', 'while', 'template', 'true', 'small',
+	'stdout', 'stdin', 'stderr', 'far', 'near', 'huge', 'requires']
 const c_reserved_chk = token.new_keywords_matcher_from_array_trie(c_reserved)
 // same order as in token.Kind
 const cmp_str = ['eq', 'ne', 'gt', 'lt', 'ge', 'le']

--- a/vlib/v/tests/structs/struct_field_name_using_c_keyword_test.v
+++ b/vlib/v/tests/structs/struct_field_name_using_c_keyword_test.v
@@ -1,0 +1,9 @@
+struct Keywords {
+	typeof u8
+}
+
+fn test_struct_field_name_using_c_reserved() {
+	key := Keywords{}
+	println(key)
+	assert true
+}

